FrontEnd/src/App.js

```diff
@@ -62,27 +66,63 @@ class App extends React.Component {
       scopes: ["user.read"]/*, "user.write"*/
     };
 
-    var self = this;
-
-    this.msalApp.loginPopup(loginRequest).then(function (loginResponse) {
+    this.msalApp.loginPopup(loginRequest)
+      .then(loginResponse => {
       //login success
-      self.setState({
-        tokenName: loginResponse.account.name,
-        tokenId: loginResponse.account.accountIdentifier
+      this.setState({
+        tokenName: loginResponse.account.name
       });
 
-      self.appInsights.setAuthenticatedUserContext(loginResponse.account.name);
-      self.appInsights.trackEvent({name: 'Login'});
-    }).catch(function (error) {
-      console.log(error);
-      self.appInsights.trackTrace({message: 'Login Error'});
+      this.appInsights.setAuthenticatedUserContext(loginResponse.account.name);
+      this.appInsights.trackEvent({name: 'Login Azure success'});
+      console.log(loginResponse);
+      return loginResponse;
+    }).then(loginResponse => {
+      console.log(loginResponse.idToken.rawIdToken);
+      return fetch(process.env.REACT_APP_SEPES_LOGIN_URL, {
+        method: "post",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({"Usename": loginResponse.account.name, "idToken": loginResponse.idToken.rawIdToken, "Expiration": "later"})
+      });
+    }).then(respnse => respnse.text())
+      .then(jwt => {
+      this.setState({tokenId: jwt});
+      localStorage.setItem(JWT_NAME, jwt);
```

Comment:
You should maybe store tokens into sessionStorage if you don't need them to be long lived. As in living longer than a browsing session.